### PR TITLE
[Diagnostics] Adjust OoO fix-it intervals to account for replacement of first argument

### PR DESCRIPTION
**Pull request by xedin**
Current logic fix-it didn't account for the first argument being re-ordered.
The start position for the first argument should always be the next token
after left paren denoting a start of the argument list.

Resolves: rdar://problem/70764991

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please smoke test

File: lib/Sema/CSDiagnostics.cpp [hunk 1]
```diff
@@ -4823,13 +4823,27 @@ bool OutOfOrderArgumentFailure::diagnoseAsError() {
...
-    auto removalRange =
-        SourceRange(Lexer::getLocForEndOfToken(
-                        SM, tuple->getElement(ArgIdx - 1)->getEndLoc()),
-                    firstRange.End);
+    SourceLoc removalStartLoc;
+    // For the first argumet it's always end of `(`.
+    if (ArgIdx == 0) {
+      removalStartLoc = tuple->getLParenLoc();
+    } else {
+      // For all argument arguments, start is next token past the previous
```
Comment (hborla) [inline lib/Sema/CSDiagnostics.cpp:4831]: Nit: I'm assuming "argument arguments" is a typo

**Review by hborla — APPROVED** [review]

File: lib/Sema/CSDiagnostics.cpp (hunk 1 above)
Comment (xedin) [reply lib/Sema/CSDiagnostics.cpp:4831]: Heh, thank you!

Comment (xedin) [reply lib/Sema/CSDiagnostics.cpp]: Fixed!

**Comment by xedin** [comment]
@swift-ci please smoke test

**Pull request merged**